src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.LogicApps.tests.ps1

```diff
@@ -678,4 +710,467 @@ InModuleScope Arcus.Scripting.LogicApps {
+            It "Doesn't disable anything when the stopType is not recognized" {
+                # Arrange
+                Mock Invoke-WebRequest {}
+                Mock Disable-AzLogicApp {}
+                Mock Get-AzCachedAccessToken -MockWith {
+                    return @{
+                        SubscriptionId = "123456"
+                        AccessToken = "accessToken"
+                    }
+                }
+
+                # Act
+                Disable-AzLogicAppsFromConfig -DeployFileName "$PSScriptRoot\Files\LogicApps\Standard\deploy-orderControl-unknownStopType.json" -ResourceGroupName "ignored-resource-group"
+
+                # Assert
+                Assert-MockCalled Get-AzCachedAccessToken -Scope It -Times 1
+                Assert-MockCalled Invoke-WebRequest -Scope It -Times 0
+                Assert-MockCalled Disable-AzLogicApp -Scope It -Exactly 0
+            }
+            It "Doesn't disable anything when both checkType & stopType is 'None'" {
+                # Arrange
+                Mock Invoke-WebRequest {}
+                Mock Disable-AzLogicApp {}
+                Mock Get-AzCachedAccessToken -MockWith {
+                    return @{
+                        SubscriptionId = "123456"
+                        AccessToken = "accessToken"
+                    }
+                }
+
+                # Act
+                Disable-AzLogicAppsFromConfig -DeployFileName "$PSScriptRoot\Files\LogicApps\Standard\deploy-orderControl-none.json" -ResourceGroupName "ignored-resource-group"
+
+                # Assert
+                Assert-MockCalled Get-AzCachedAccessToken -Scope It -Times 1
+                Assert-MockCalled Invoke-WebRequest -Scope It -Times 0
+                Assert-MockCalled Disable-AzLogicApp -Scope It -Exactly 0
+            }
+            It "Doesn't disable anything when checkType = NoWaitingOrRunningRuns but returns a running run for unknown stopType" {
+                # Arrange
+                $script:i = 0
+                $resourceGroup = "my-resource-group"
+                $logicAppNames = @("snd-async", "ord-sthp-harvest-order-doublechecker", "ord-sthp-harvest-order-doublechecker")
+
+                Mock Invoke-WebRequest -MockWith {
+                    $script:i++
+                    if ($script:i -gt 2) {
+                        Write-Host "Returning empty (no running, no waiting) runs"
+                        return [pscustomobject] @{
+                            Content = '{"value":[]}'
+                        }
+                    } else {
+                        Write-Host "Returning 1 running run"
+                        return [pscustomobject] @{
+                            Content = '{"value":[{"properties":{"status":"Running"}}]}'
+                        }
+                    }
+                }
+                Mock Disable-AzLogicApp {}
+                Mock Get-AzCachedAccessToken -MockWith {
+                    $Global:subscriptionId = '123456'
```

Comment:
Thought that those global variables were for testing... It feels a bit an anti-pattern to have them there as global ones to use them here. Thinking maybe even a security risk that other following scripts could also access them? Isn't there another way we can stub out some function and return them? Or maybe create `$script:` variables here and store them once this is called?

Reply:
Not entirely sure what you mean, but this is only used on the tests. 
This was really the only way I could get the global variable populated if I remember correctly. But if there is a better solution I am all for it 👍🏻

Reply:
Yes, my concern is that this was only used for the tests, and that it could potentially be vulnerable. I'm wondering why we need it, though, as the `Get-AzCachedAccessToken` function already returns the same information? And once it is set, it is not used for assertion, too.